app/controllers/state_file/questions/ny_county_controller.rb

```diff
@@ -4,24 +4,9 @@ module StateFile
   module Questions
     class NyCountyController < AuthenticatedQuestionsController
       include ReturnToReviewConcern
-      
-      def edit
-        @counties = counties_from_csv
-        super
-      end
 
       private
 
-      def counties_from_csv
-        csv_file_path = Rails.root.join('docs', 'ny_school_districts.csv')
-        csv_content = File.read(csv_file_path)
-
-        io = StringIO.new(csv_content)
-        CSV.parse(io, headers: true).map do |row|
-          row['County']
-        end.uniq
-      end
-
```

Comment:
moved to application_helper

Reply:
Non-blocking preference: I appreciate the effort to refactor, my instinct is that this would be clearer in the controller, because the data is specific to this particular page. I wouldn't put this in application_helper unless I expected it to be used by multiple controllers, ideally only if used by many of them, because it makes the method available to all controllers in the app. Here's another option, for your consideration:

```ruby
class NyCountyController < AuthenticatedQuestionsController
  include ReturnToReviewConcern
  helper_method :counties_from_csv
  # ...
  def counties_from_csv
    # ...
  end
end
```

I'd also recommend checking out `config/initializers/nys_data.rb` and `app/lib/efile/ny/constants.rb`. The `counties_from_csv` method, as currently implemented, will load the same data from the file system with every new request. If instead we load the options from the filesystem during the preload process, they can hang out in memory and won't need to be reloaded.

Reply:
i did this because it doesn't work in the controller, the @counties variable is lost when validation fails. i assume that's solvable somehow but it wasn't immediately obvious to me, and when i started to poke around i noticed that we frequently use this pattern to load options from application_helper instead of in controllers. 

i didn't implement the csv loading code, but the caching sounds useful, i'll look into that before i merge.